### PR TITLE
Match Remaining En_Niw Functions (En_Niw OK)

### PR DESCRIPTION
**Pull request by engineer124**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Niw/z_en_niw.c [hunk 1]
```diff
@@ -324,18 +331,19 @@ void EnNiw_SetupIdle(EnNiw* this) {
...
 
-#ifdef NON_MATCHING
-// non-matching: stack offset and regalloc
-// EnNiw_Idle
-void func_808919E8(EnNiw* this, GlobalContext* globalCtx) {
+void EnNiw_Idle(EnNiw* this, GlobalContext* globalCtx) {
+    f32 var_x2;
+    f32 var_z2;
+    f32 var_x1;
+    f32 var_z1;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Niw/z_en_niw.c]:
```suggestion
    f32 var_x1 = randPlusMinusPoint5Scaled(100.0f);
    f32 var_z1 = randPlusMinusPoint5Scaled(100.0f);
```

**Review by EllipticEllipsis — APPROVED** [review]

File: src/overlays/actors/ovl_En_Niw/z_en_niw.c [hunk 2]
```diff
@@ -369,24 +377,20 @@ void func_808919E8(EnNiw* this, GlobalContext* globalCtx) {
...
+                var_x1 += 100.0f;
             }
-            if (newPos.z < 0.0f) {
-                newPos.z -= 100.0f;
+            if (var_z1 < 0.0f) {
+                var_z1 -= 100.0f;
             } else {
-                newPos.z += 100.0f;
+                var_z1 += 100.0f;
             }
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Niw/z_en_niw.c]: Would the `CLAMP` macro match here?

File: src/overlays/actors/ovl_En_Niw/z_en_niw.c [hunk 3]
```diff
@@ -656,27 +655,21 @@ void func_808924B0(EnNiw* this, GlobalContext* globalCtx) {
...
-        this->unk2A4.y = this->actor.world.pos.y;
-        this->unk2A4.z = this->actor.world.pos.z;
-        this->unk2B0.x = this->actor.world.pos.x;
-        this->unk2B0.y = this->actor.world.pos.y;
-        this->unk2B0.z = this->actor.world.pos.z;
-
-        this->unkTimer252 = this->unkTimer250 = this->unk298;
+        this->unk2A4.x = this->unk2B0.x = this->actor.world.pos.x;
+        this->unk2A4.y = this->unk2B0.y = this->actor.world.pos.y;
+        this->unk2A4.z = this->unk2B0.z = this->actor.world.pos.z;
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Niw/z_en_niw.c:660]: This one hurts, this one should have been obvious, damn it.

File: src/overlays/actors/ovl_En_Niw/z_en_niw.c (hunk 2 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Niw/z_en_niw.c]: The `+=` and `-=` make me think not

**Review by hensldm — COMMENTED** [review]
Nice. Just a few minor possible documentation comments

File: src/overlays/actors/ovl_En_Niw/z_en_niw.c [hunk 4]
```diff
@@ -656,27 +652,21 @@ void func_808924B0(EnNiw* this, GlobalContext* globalCtx) {
     // and it does it BEFORE the if block, this is just unoptimized.
```
[the remark this answers is not in the document]
Comment (isghj5) [reply src/overlays/actors/ovl_En_Niw/z_en_niw.c:652]: My bad, I left that there out of frustration of the dev, but yeah it should be removed.

**Review by isghj5 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]
🐔

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**